api/explore/explore.js

```diff
@@ -20,14 +24,59 @@ function randomChoice(arr) {
 
 async function innerSearch(searchString, amount, request) {
   const bool = {
-    must: {
+    must: [{
       simple_query_string: {
         query: searchString.replace(' |_', '+'),
         fields: ['title', 'ingredients', 'tags'],
       },
-    },
+    }],
   };
 
+  try {
+    const user = await auth.checkJwt(request);
+    if (user) {
+      await User.findById(user.sub, 'excludeTerms diet', (err, prefs) => {
+        if (err) throw (err);
+        if (prefs.excludeTerms.length > 0) {
+          bool.must_not = prefs.excludeTerms.map((term) => ({ match: { ingredients: { query: term, fuzziness: 'AUTO:0,4' } } }));
+        }
+        if (prefs.diet.length > 0) {
+          const dietQueryPart = {
+            bool: {
+              must: [],
+            },
+          };
+          prefs.diet.forEach((item) => {
+            if (!get(item, 'tags', null)) {
+              dietQueryPart.bool.must.push(
+                {
+                  bool: {
+                    must: [
+                      { match: { tags: item } },
+                    ],
+                  },
+                },
+              );
+            } else {
+              dietQueryPart.bool.must.push(
+                {
+                  bool: {
+                    should: item.tags.map((tag) => ({ match: { tags: tag } })),
```

Comment:
This should be moved to searchFunc there is already a jwt check there, you just need to move it before the ES search and use user after to mark the saved ones as well